plugins/sy-personio-jobs-source-plugin/personio-jobs-api.js

```diff
@@ -0,0 +1,137 @@
+ * so we store data in custom fields that are actually
+ * meant as content.
+ *
+ * We find the matching descriptions for the keys
+ * and return their values. The receiver can the easily destructure it:
+ *
+ * const [slug, short] = extractDescriptionValues(jobDescriptions, [slugKey, shortDescriptionKey])
+ */
+const extractDescriptionValues = (descriptions, keys) => {
+  return keys.reduce((memo, key) => {
+    const description = descriptions.find((item) => item.name.trim() === key);
+
+    if (!description) {
+      return memo;
+    }
+
+    const value = striptags(description.value).trim();
+    memo.push(value);
+
+    return memo;
+  }, []);
+};
+
+/*
+   Drop descriptions. We mainly use that
+   to remove meta data descriptions that we 
+   are going to extract in beforehand.
+ */
+function dropDescriptions(descriptions) {
+  return descriptions.reduce((memo, item) => {
+    if (
+      [PERSONIO_SLUG_FIELD_NAME, PERSONIO_SHORT_DESCRIPTION_NAME].includes(
+        item.name,
+      )
+    ) {
+      return memo;
+    }
+
+    memo.push(item);
+    return memo;
+  }, []);
+}
+
+/**
+ * Incoming an array of object with the format of:
+ * {name, value} where name is the custom headline defined in Personio
+ * and value is a rich text fields that includes html tag.
+ *
+ * This function makes sure that we strip and trim the content
+ * which is given to us in a pretty raw form.
+ *
+ * In addition we rename the fields so we can forward them as is into
+ * the Gatsby store.
+ */
+function parseDescriptions(jobDescriptions) {
+  const [slug, short] = extractDescriptionValues(jobDescriptions, [
+    PERSONIO_SLUG_FIELD_NAME,
+    PERSONIO_SHORT_DESCRIPTION_NAME,
+  ]);
+  const remainingDescriptions = dropDescriptions(jobDescriptions, [
+    PERSONIO_SLUG_FIELD_NAME,
+    PERSONIO_SHORT_DESCRIPTION_NAME,
+  ]);
+
+  const sections = remainingDescriptions.map((item) => ({
+    headline: item.name,
+    descriptionHtml: item.value.trim(),
+    description: striptags(item.value.trim()),
+  }));
+
+  return { sections, slug, short };
+}
+
+/**
+ * Given a set of positions from the XML API of Personio
+ * we need to do some processing to have a proper dataset
+ * we can feed into the Gatsby GraphQL backend.
+ */
+function normalize(positions, lang) {
+  return positions.reduce((memo, position) => {
+    const { jobDescription } = position.jobDescriptions;
+
+    if (!jobDescription) {
+      console.warn(
+        `No description for job "${position.name}" (${position.id}) found! Job ignored.`,
+      );
+      return memo;
+    }
+
+    const { sections, slug, short } = parseDescriptions(jobDescription);
+    delete position.jobDescriptions;
+
+    memo.push({
+      ...position,
+      slug,
+      short,
+      sections,
+      lang,
+    });
+
+    return memo;
+  }, []);
+}
+
+/**
+ * Fetch all available jobs for the given language
+ * from our Personio API endpoint.
+ */
+async function fetch(language) {
```

Comment:
would call it `fetchJobPositions` instead of a plain fetch, as it could be misunderstood with the global fetch

Reply:
I had it like that but I though the package tells the story so no misunderstanding. Let's keep it as is and we can still refactor if continues to upset us 😁